File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/ClassResizableTextField.java
package io.github.notenoughupdates.moulconfig.gui;

import io.github.notenoughupdates.moulconfig.gui.component.TextFieldComponent;
import io.github.notenoughupdates.moulconfig.observer.GetSetter;

public class ClassResizableTextField extends TextFieldComponent {
    private int width = 20;

    public ClassResizableTextField(GetSetter<String> text) {
        super(text, 20);
    }

    public void setWidth(int width) {
        this.width = width;
    }

    @Override
    public int getWidth() {
        return width;
    }

    @Override
    public void render(GuiImmediateContext context) {
        super.render(context.translated(0, 0, width, 18));
    }

    @Override
    public boolean mouseEvent(MouseEvent mouseEvent, GuiImmediateContext context) {
        return super.mouseEvent(mouseEvent, context.translated(0, 0, width, 18));
    }

    @Override
    public boolean keyboardEvent(KeyboardEvent event, GuiImmediateContext context) {
        return super.keyboardEvent(event, context.translated(0, 0, width, 18));
    }
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/ClassResizableTextField.kt
package io.github.notenoughupdates.moulconfig.gui

import io.github.notenoughupdates.moulconfig.gui.component.TextFieldComponent
import io.github.notenoughupdates.moulconfig.observer.GetSetter

class ClassResizableTextField(text: GetSetter<String>) : TextFieldComponent(
    text,
    20,
) {
    private var width = 20

    fun setWidth(width: Int) {
        this.width = width
    }

    override fun getWidth(): Int {
        return width
    }

    override fun render(context: GuiImmediateContext) {
        super.render(context.translated(0, 0, width, 18))
    }

    override fun mouseEvent(mouseEvent: MouseEvent, context: GuiImmediateContext): Boolean {
        return super.mouseEvent(mouseEvent, context.translated(0, 0, width, 18))
    }

    override fun keyboardEvent(event: KeyboardEvent, context: GuiImmediateContext): Boolean {
        return super.keyboardEvent(event, context.translated(0, 0, width, 18))
    }

}